diff --git a/vendor/magento/module-ui/view/frontend/web/js/view/messages.js b/vendor/magento/module-ui/view/frontend/web/js/view/messages.js
index e6d3bc8..ddb9a99 100644
--- a/vendor/magento/module-ui/view/frontend/web/js/view/messages.js
+++ b/vendor/magento/module-ui/view/frontend/web/js/view/messages.js
@@ -66,9 +66,10 @@
 
             // Hide message block if needed
             if (isHidden) {
+                $(window).scrollTop(0);
                 setTimeout(function () {
                     $(self.selector).hide('blind', {}, 500);
-                }, 5000);
+                }, 30000);
             }
         }
     });
