diff --git a/vendor/magento/module-checkout/Model/GuestPaymentInformationManagement.php b/vendor/magento/module-checkout/Model/GuestPaymentInformationManagement.php
index e6d3bc8..ddb9a99 100644
--- /vendor/magento/module-checkout/Model/GuestPaymentInformationManagement.php	2018-09-04 16:53:44.000000000 -0400
+++ /vendor/magento/module-checkout/Model/GuestPaymentInformationManagement.php	2019-01-28 18:47:36.848064426 -0500
@@ -112,7 +112,7 @@
             } catch (\Exception $e) {
                 $this->getLogger()->critical($e);
                 throw new CouldNotSaveException(
-                    __('An error occurred on the server. Please try to place the order again.'),
+                    __('Credit card error. Check billing zipcode and card infromation.'),
                     $e
                 );
             }
