diff --git a/vendor/magento/module-elasticsearch/Model/ResourceModel/Index.php b/vendor/magento/module-elasticsearch/Model/ResourceModel/Index.php
index 59519eb..5d61717 100644
--- a/vendor/magento/module-elasticsearch/Model/ResourceModel/Index.php
+++ b/vendor/magento/module-elasticsearch/Model/ResourceModel/Index.php
@@ -83,6 +83,15 @@ class Index extends \Magento\AdvancedSearch\Model\ResourceModel\Index
                         $value = $indexData[$attribute->getAttributeId()][$productId];
                     } else {
                         $value = implode(' ', $indexData[$attribute->getAttributeId()]);
+                        $backendType = $attribute->getBackendType();
+                        if ($backendType == 'datetime' || $backendType == 'decimal'|| $backendType == 'int') {
+                            $value = array_unique($indexData[$attribute->getAttributeId()], SORT_STRING);
+                            if (count($value) === 1) {
+                                $value = array_shift($value);
+                            } else {
+                                $value = null;
+                            }
+                        }
                     }
                 } else {
                     $value = $indexData[$attribute->getAttributeId()];
