diff --git a/vendor/magento/module-checkout/Model/PaymentInformationManagement.php b/vendor/magento/module-checkout/Model/PaymentInformationManagement.php
index e6d3bc8..ddb9a99 100644
--- /vendor/magento/module-checkout/Model/PaymentInformationManagement.php	2018-09-04 16:53:44.000000000 -0400
+++ /vendor/magento/module-checkout/Model/PaymentInformationManagement.php	2019-01-28 18:47:49.418346426 -0500
@@ -89,7 +89,7 @@
         } catch (\Exception $e) {
             $this->getLogger()->critical($e);
             throw new CouldNotSaveException(
-                __('An error occurred on the server. Please try to place the order again.'),
+                __('Credit card error. Check billing zipcode and card infromation.'),
                 $e
             );
         }
