diff --git a/vendor/magento/framework/Escaper.php b/vendor/magento/framework/Escaper.php
index e6d3bc8..ddb9a99 100644
--- a/vendor/magento/framework/Escaper.php	2018-11-19 11:58:50.000000000 -0500
+++ b/vendor/magento/framework/Escaper.php	2019-05-21 16:14:12.353300488 -0400
@@ -77,7 +77,7 @@
                         throw new \Exception($errorString, $errorNumber);
                     }
                 );
-                $string = mb_convert_encoding($data, 'HTML-ENTITIES', 'UTF-8');
+                $string = mb_convert_encoding(preg_replace('/&/', '&amp;', html_entity_decode($data)), 'HTML-ENTITIES', 'UTF-8');
                 try {
                     $domDocument->loadHTML(
                         '<html><body id="' . $wrapperElementId . '">' . $string . '</body></html>'
